Add whole-words? option for generated patterns to match only whole words

joelittlejohn opened the pull request:
This option causes patterns to require a word boundary on either end of the match. For instance a pattern built like:

```clj
(frak/pattern ["foo" "bar" "baz"] {:whole-words? true})
```

Will create a match against the following strings:

    "foo is a metasyntactic variable"
    "sometimes we use baz"

but will not create a match against the following:

    "food and nutrition"
    "clubs and bars"

noprompt commented on src/cljx/frak.cljx:
```diff
@@ -91,7 +96,9 @@
         (string/join strs)
-        (re-group-end optional?)))))
+        (re-group-end optional?)
+        (if *whole-words*
```
`when`

noprompt commented on test/frak_test.clj:
```diff
@@ -71,3 +71,9 @@
     ["aching" "achingly"]))
+
+(deftest pattern-whole-words
+  (is (= ["k pop"] (re-seq (pattern ["pop" "k pop"]) "uk pop")))
```
I would prefer a newline between the expected and actual values here.

noprompt commented:
Sorry for the late response! I like this. If you could address the two small issues I referenced in the code I'll get this merged. If you're busy I can address it myself a little later. Thanks!

joelittlejohn commented:
Great, thanks for looking at this Joel! I've addressed the comments. Any other suggestions please shout. Glad to be able to contribute to this fantastic library.

noprompt commented:
@joelittlejohn I've pushed a new release. It's been a long time since I've built this project so let me know if there are any issues.

joelittlejohn commented:
@noprompt brilliant, thanks!

joelittlejohn commented:
I've integrated the latest version in a project I'm working on and everything looks good :+1:
